src/app/shared/live-region/live-region.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+import { environment } from '../../../environments/environment';
+import { UUIDService } from '../../core/shared/uuid.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LiveRegionService {
+
+  constructor(
+    protected uuidService: UUIDService,
+  ) {
+  }
+
+  /**
+   * The duration after which the messages disappear in milliseconds
+   * @protected
+   */
+  protected messageTimeOutDurationMs: number = environment.liveRegion.messageTimeOutDurationMs;
+
+  /**
+   * Array containing the messages that should be shown in the live region,
+   * together with a uuid, so they can be uniquely identified
+   * @protected
+   */
+  protected messages: { message: string, uuid: string }[] = [];
+
+  /**
+   * BehaviorSubject emitting the array with messages every time the array updates
+   * @protected
+   */
+  protected messages$: BehaviorSubject<string[]> = new BehaviorSubject([]);
+
+  /**
+   * Whether the live region should be visible
+   * @protected
+   */
+  protected liveRegionIsVisible: boolean = environment.liveRegion.isVisible;
+
+  /**
+   * Returns a copy of the array with the current live region messages
+   */
+  getMessages(): string[] {
+    return this.messages.map(messageObj => messageObj.message);
+  }
+
+  /**
+   * Returns the BehaviorSubject emitting the array with messages every time the array updates
+   */
+  getMessages$(): BehaviorSubject<string[]> {
+    return this.messages$;
+  }
+
+  /**
+   * Adds a message to the live-region messages array
+   * @param message
+   * @return The uuid of the message
+   */
+  addMessage(message: string): string {
+    const uuid = this.uuidService.generate();
+    this.messages.push({ message, uuid });
+    setTimeout(() => this.clearMessageByUUID(uuid), this.messageTimeOutDurationMs);
```

Comment:
Another quick thing: I think using a reactive approach and a setTimeout together is kinda awkward. Using a rxjs timer() might be best, in order to keep everything reactive!

Reply:
Do you suggest simply changing that `setTimeout` line with `timer(this.messageTimeOutDurationMs).subscribe(() => this.clearMessageByUUID(uuid));`?
I don't see how that makes a big difference. If anyone else prefers using `timer` over `setTimeout` I'll make the change. 
In the current codebase `setTimeout` is used more frequently than `timer`, but that shouldn't be a stopper if `timer` is an actual improvement over `setTimeout`.

Reply:
I see your reasoning, and I'm ok with keeping it consistent with the current codebase!